url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_AgeGenderLineChart.java

```diff
@@ -0,0 +1,870 @@
+            int startAgeOfEndYearInterval,
+            int decimalPlacePrecisionForCalculations,
+            int decimalPlacePrecisionForDisplay,
+            RoundingMode aRoundingMode) {
+        init(
+                executorService,
+                file,
+                format,
+                title,
+                dataWidth,
+                dataHeight,
+                xAxisLabel,
+                yAxisLabel,
+                drawOriginLinesOnPlot,
+                ageInterval,
+                startAgeOfEndYearInterval,
+                decimalPlacePrecisionForCalculations,
+                decimalPlacePrecisionForDisplay,
+                aRoundingMode);
+    }
+
+    public static void main(String[] args) {
+        /*
+         * Initialise title and File to write image to
+         */
+        String title;
+        File file;
+        String format = "PNG";
+        if (args.length != 2) {
+            System.out.println(
+                    "Expected 2 args:"
+                    + " args[0] title;"
+                    + " args[1] File."
+                    + " Recieved " + args.length + " args.");
+            // Use defaults
+            title = "Age Gender Population Line Chart";
+            System.out.println("Use default title: " + title);
+            file = new File(
+                    new File(System.getProperty("user.dir")),
+                    title.replace(" ", "_") + "." + format);
+            System.out.println("Use default File: " + file.toString());
+        } else {
+            title = args[0];
+            file = new File(args[1]);
+        }
+        int dataWidth = 1000;//250;
+        int dataHeight = 500;
+        String xAxisLabel = "Population";
+        String yAxisLabel = "Age";
+        boolean drawOriginLinesOnPlot = true;
+        int ageInterval = 1;
+        int startAgeOfEndYearInterval = 90;//95;
+        int decimalPlacePrecisionForCalculations = 10;
+        int decimalPlacePrecisionForDisplay = 3;
+        RoundingMode aRoundingMode = RoundingMode.HALF_UP;
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        Generic_AgeGenderLineChart plot = new Generic_AgeGenderLineChart(
+                executorService,
+                file,
+                format,
+                title,
+                dataWidth,
+                dataHeight,
+                xAxisLabel,
+                yAxisLabel,
+                drawOriginLinesOnPlot,
+                ageInterval,
+                startAgeOfEndYearInterval,
+                decimalPlacePrecisionForCalculations,
+                decimalPlacePrecisionForDisplay,
+                aRoundingMode);
+        plot.setData(plot.getDefaultData());
+        plot.run();
+    }
+
+    @Override
+    public void drawData() {
+        //drawLineChartUsingMeanAndStandardDeviation();
+        drawLineChartUsingMinQ1MedianQ3Max();
+    }
+
+    public void drawLineChartUsingMeanAndStandardDeviation() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 59 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 215-266 into a separate method.